network/p2p/libp2pNode.go

```diff
@@ -528,32 +572,35 @@ func bootstrapLibP2PHost(ctx context.Context,
 	// gather all the options for the libp2p node
 	options := []config.Option{
 		libp2p.ListenAddrs(sourceMultiAddr), // set the listen address
-		libp2p.Identity(key),                // pass in the networking key
-		libp2p.ConnectionManager(conMgr),    // set the connection manager
+		libp2p.Identity(libp2pKey),          // pass in the networking key
 		transport,                           // set the protocol
-		libp2p.Ping(true),                   // enable ping
+		libp2p.Ping(pingEnabled),            // enable ping
 	}
 
-	// if allowlisting is enabled, create a connection gator with allowListAddrs
-	if allowList {
-		// create a connection gater
-		connGater = newConnGater(logger)
+	return options, nil
+}
 
-		// provide the connection gater as an option to libp2p
-		options = append(options, libp2p.ConnectionGater(connGater))
-	}
+// WithDefaultPubSub returns a NodeOption that initializes a GossipSub object for the node libp2p host with the given
+// options
+func WithDefaultPubSub(psOption ...pubsub.Option) NodeOption {
+	return func(node *Node) error {
 
-	// create the libp2p host
-	libP2PHost, err := libp2p.New(ctx, options...)
-	if err != nil {
-		return nil, nil, nil, fmt.Errorf("could not create libp2p host: %w", err)
+		// Creating a new PubSub instance of the type GossipSub with psOption
+		pubSub, err := pubsub.NewGossipSub(node.ctx, node.host, psOption...)
```

Comment:
libp2p provides 3 [pubsub](https://github.com/libp2p/go-libp2p-pubsub) mechanism `floodsub`, `gossipsub`, `randomsub`.  We use `gossipSub` as our default